src/lib/utils/board.js

```diff
@@ -0,0 +1,16 @@
+export const removeBoardItem = (boards, nodeId) =>
+  boards.map((board) => ({
+    ...board,
+    items: board.items.filter((item) => item.id !== nodeId),
+  }));
+
+export const insertBoardItem = (boards, parentId, item, targetIndex) =>
```

Comment:
P3: 요건 가벼운 제안 정도로 봐주세요! map 메서드안에서 변경 및 복사를 거치는 방식보다 읽기 편하고 성능상 더 나은 방법으로 변경해보았어요. 성능상 차이는 작아서 구경정도로 봐주세요~

```suggestion

export const insertBoardItem = (boards, parentId, item, targetIndex) => {
  const targetBoardIndex = boards.findIndex((board) => board.id === parentId);
  const targetBoard = boards[targetBoardIndex];
  const newItems = [...targetBoard.items];
  newItems.splice(targetIndex, 0, item);
  const newBoard = { ...targetBoard, items: newItems };
  const newBoards = [...boards];
  newBoards[targetBoardIndex] = newBoard;
  return newBoards;
};

```

Reply:
> P3: 요건 가벼운 제안 정도로 봐주세요! map 메서드안에서 변경 및 복사를 거치는 방식보다 읽기 편하고 성능상 더 나은 방법으로 변경해보았어요. 성능상 차이는 작아서 구경정도로 봐주세요~

targetBoard 정보를 가져와서 처리하는 방식이네요!
map안에서 다 처리하기보다 조금은 분리해서 나누어보는것도 연습해보아야겠어요.
감사합니다. 🙇‍♂️